src/main/java/httpConnectionProcessing/HttpConnectionToProcess.java

```diff
@@ -26,22 +28,28 @@ public void execute() {
     }
   }
 
-  private IHandler getHandler() throws BadConnectionException {
-    HttpRequest request;
-    IHandler handler;
+  private HttpRequest getRequest() throws BadConnectionException {
     try {
-      request = httpRequestBuilder.getNextRequest(client.clientSocketInput());
-      handler = handlerRouter.selectHandler(request);
+      return httpRequestBuilder.getNextRequest(client.clientSocketInput());
     }
     catch (InvalidHttpRequestException e) {
       e.printStackTrace();
+      return null;
+    }
+  }
+
+  private IHandler getHandler(HttpRequest request) throws BadConnectionException {
+    IHandler handler;
+    if (request == null) {
```

Comment:
@dadarek Looks like previously `getRequest` was not a separate method and instead was inlined as part of `getHandler`. The request could be invalid, in which case an exception was raised and caught, leading to returning the "Bad Request" handler. However, now the request is grabbed first and passed to this method. But the original HTTP request could still be invalid, and probably shouldn't be coerced into an `HttpRequest` object if it's gibberish...